src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -226,6 +228,14 @@ private synchronized void processCISShadow(String version, Map<String, Object> d
                         .log("Retry workflow for getting connectivity info interrupted");
                 Thread.currentThread().interrupt();
                 return;
+            } catch (ValidationException | ResourceNotFoundException e) {
```

Comment:
why is this changing? Can't we just change: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/2eb6d62da250522efd48a18aa7b67a0ed1b757a8/src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/ConnectivityInformation.java#L96 to only cache when successful?

Reply:
trying to prevent unnecessary rotation if `ValidationException/ResourceNotFoundException` is thrown by CIS.  if that doesn't bubble up, then CISShadowMonitor will see empty list and rotate the certs

Reply:
updating the reported state here seems unusual. Why is it the right thing to do?

Reply:
it doesn't have an effect on processing, i think it's more of a way to tell that the shadow was processed/dealt with just by looking at the shadow contents itself?   so for debugging purposes, and to keep with the convention that CISShadowMontior updates the shadow when it's finishing processing a shadow.